main.py

```diff
@@ -558,6 +563,17 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
+
+        # Even if args.do_target_training = 0, evaluate on diagnostic tasks.
```

Comment:
This comment seems misleading. We should evaluate on all the target_tasks when do_full_eval is set, whether or not they're diagnostics.

Reply:
In that case we can actually cut a lot of the logic here if we're evaluating on all target_tasks.

Reply:
Yep!